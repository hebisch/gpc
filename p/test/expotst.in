1.0e-23
1.0e-17
0.1
0.0
10.0
1.0e+17
1.0e+23
