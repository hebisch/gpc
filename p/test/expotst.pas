program test(output);
var
x :real;

function expo(a:real):integer;
begin
  if a=0 then
    expo:=0
  else
    expo:=round(ln(abs(a))/ln(10)-0.49999)
end;

begin
while not eof do begin
        readln(x);
        writeln(expo(x):10)
        end
end.
