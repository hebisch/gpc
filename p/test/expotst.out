       -23
       -17
        -1
         0
         1
        17
        23
